module/zfs/dmu_redact.c

```diff
@@ -823,6 +823,12 @@ perform_thread_merge(bqueue_t *q, uint32_t num_threads,
 	kmem_free(redact_nodes, num_threads * sizeof (*redact_nodes));
 	if (current_record != NULL)
 		bqueue_enqueue(q, current_record, sizeof (current_record));
+
+	for (int i = 0; i < num_threads; i++) {
+		struct redact_thread_arg *targ = &thread_args[i];
+		bqueue_destroy(&targ->q);
+	}
```

Comment:
It looks to me like we don't need this chunk.  In `dmu_redact_snap()->perform_redaction()` we should wait until the `redact_merge_thread` completes at which point we'll exit `perform_redaction()` and destroy the bqueue with you other addition.  Though I could be mistaken, @pcd1193182 can you confirm this?

Reply:
This should work, though I would move it up into the previous for loop where we call avl_remove for each worker thread.

This is destroying the worker threads' bqueues, which are different from the queue from the merge thread to the main thread, so we do need to have a separate destroy call.

Reply:
@pcd1193182 thanks for taking a look at this.  @lundman if you can refresh the patch to incorporate Paul's feedback and rebase it on master we can get this merged.